### PR TITLE
Support Fast DDS 2.x and both versions of Fast CDR [21146]

### PR DESCRIPTION
**Pull request by richiware**
Added support for working with latests 2.x versions of Fast DDS and both versions of Fast CDR.

Depends on:
- eprosima/idl-parser#147

### CODE REVIEW

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterfacev1.stg [hunk 1]
```diff
@@ -0,0 +1,1747 @@
+/*************************************************************************
+ * Copyright (c) 2012 eProsima. All rights reserved.
+ *
+ * This copy of FASTRPC is licensed to you under the terms described in the
+ * FASTRPC_LICENSE file included in this distribution.
+ *
+ *************************************************************************/
```
Comment (MiguelCompany) [inline fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterfacev1.stg:7]: Format

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperationv1.stg [hunk 2]
```diff
@@ -0,0 +1,1301 @@
+/*************************************************************************
+ * Copyright (c) 2012 eProsima. All rights reserved.
+ *
+ * This copy of FASTRPC is licensed to you under the terms described in the
+ * FASTRPC_LICENSE file included in this distribution.
+ *
+ *************************************************************************/
```
Comment (MiguelCompany) [inline fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperationv1.stg:7]: Format

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TypesCdrAuxHeaderImpl.stg [hunk 3]
```diff
@@ -0,0 +1,494 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (MiguelCompany) [inline fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TypesCdrAuxHeaderImpl.stg:1]:
```suggestion
// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: include/transports/dds/ProxyTransport.h [hunk 4]
```diff
@@ -23,151 +23,176 @@ class DDSDomainParticipant;
...
+     */
+    RPC_DllAPI ProxyTransport(
+            const char* const remoteServiceName = NULL,
+            const char* const instanceName = NULL,
+            int domainId = 0,
+            long milliseconds = 10000L);
+
+private:
+
+    //TODO TEMPORAL
```
Comment (MiguelCompany) [inline include/transports/dds/ProxyTransport.h:157]: ??

File: include/transports/dds/RTPSProxyTransport.h [hunk 5]
```diff
@@ -19,118 +19,145 @@
...
+            const char* readertopicname,
+            RTPSTransport::Create_data create_data,
+            RTPSTransport::Copy_data copy_data,
+            RTPSTransport::Destroy_data destroy_data,
+            RTPSTransport::ProcessFunc processFunc,
+            size_t dataSize);
+
+private:
+
+    //TODO TEMPORAL
```
Comment (MiguelCompany) [inline include/transports/dds/RTPSProxyTransport.h:124]: ??

File: include/transports/dds/components/RTPSServerProcedureEndpoint.h [hunk 6]
```diff
@@ -22,139 +22,151 @@
 
 #include <string>
 
-namespace boost
+namespace boost {
+class mutex;
+} // namespace boost
```
Comment (MiguelCompany) [inline include/transports/dds/components/RTPSServerProcedureEndpoint.h:27]: Is there a need for a boost mutex, could we use a std::mutex instead?
Are there other boost dependencies?
Can we remove dependency on boost?

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: examples/C++/rpcdds/c++11Types/RTPSCalculator/CMakeLists.txt [hunk 7]
```diff
@@ -31,9 +31,9 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR
     endif()
 endif()
 
-set(RTPSCALCULATOR_COMMON_SOURCES_CXX CalculatorTopicsPlugin.cxx CalculatorTopics.cxx CalculatorDDSProtocol.cxx CalculatorDDSAsyncSupport.cxx)
+set(RTPSCALCULATOR_COMMON_SOURCES_CXX CalculatorTopicsPlugin.cxx CalculatorTopics.cxx CalculatorTopicsv1.cxx CalculatorDDSProtocol.cxx CalculatorDDSAsyncSupport.cxx)
```
[the remark this answers is not in the document]
Comment (richiware) [reply examples/C++/rpcdds/c++11Types/RTPSCalculator/CMakeLists.txt:30]: We decided this is out of scope of this PR.

File: examples/C++/rpcdds/c++11Types/RTPSCalculator/CalculatorDDSProtocol.cxx [hunk 8]
```diff
@@ -213,33 +233,43 @@ void CalculatorProtocol::Calculator_serve(eprosima::rpc::protocol::Protocol &pro
...
+    };
 }
-int32_t CalculatorProtocol::Calculator_addition(/*in*/ int32_t value1, /*in*/ int32_t value2)
+
+int32_t
+ CalculatorProtocol::Calculator_addition(/*in*/ 
+int32_t
+ value1, /*in*/ 
+int32_t
+ value2)
```
[the remark this answers is not in the document]
Comment (richiware) [reply examples/C++/rpcdds/c++11Types/RTPSCalculator/CalculatorDDSProtocol.cxx]: Done in 44e1b980

File: examples/C++/rpcdds/c++11Types/RTPSCalculator/Calculator.h [hunk 9]
```diff
@@ -1,74 +1,109 @@
...
 #include <string>
 #include <vector>
-#include <map>
-#include <bitset>
+
+#include <fastcdr/cdr/fixed_size_string.hpp>
+#include <fastcdr/xcdr/external.hpp>
+#include <fastcdr/xcdr/optional.hpp>
+
+// TODO Poner en el contexto.
```
[the remark this answers is not in the document]
Comment (richiware) [reply examples/C++/rpcdds/c++11Types/RTPSCalculator/Calculator.h]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/parser/typecode/AliasTypeCode.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/parser/typecode/AliasTypeCode.java]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/Common.stg [hunk 11]
```diff
@@ -20,11 +20,11 @@ paramDeclsByRef(params, separatorInicial) ::= <<
 $if(params)$$separatorInicial$$endif$$params : {param |/*$param.comment$*/ $paramTypeByRef(typecode=param.typecode)$ $param.name$}; separator=", "$
 >>
 
-extractParams(params, separatorInicial) ::= <<
+extractParams(params, separatorInicial="") ::= <<
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/Common.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByInterfacev1.stg [hunk 12]
```diff
@@ -0,0 +1,1003 @@
+/*************************************************************************
+ * Copyright (c) 2013 eProsima. All rights reserved.
+ *
+ * This copy of FASTRPC is licensed to you under the terms described in the
+ * FASTRPC_LICENSE file included in this distribution.
+ *
+ *************************************************************************/
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByInterfacev1.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByInterfacev1.stg [hunk 13]
```diff
@@ -0,0 +1,1003 @@
...
+
+#include "$ctx.filename$.h"
+#include <$ctx.includePrefix$protocols/dds/MessageHeader.h>
+#include <$ctx.includePrefix$utils/Messages.h>
+
+#include <cstddef>
+
+#if defined(_WIN32)
+#if defined(EPROSIMA_USER_DLL_EXPORT)
+#define user_cpp_DllExport __declspec( dllexport )
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByInterfacev1.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperationv1.stg [hunk 14]
```diff
@@ -0,0 +1,768 @@
+/*************************************************************************
+ * Copyright (c) 2013 eProsima. All rights reserved.
+ *
+ * This copy of FASTRPC is licensed to you under the terms described in the
+ * FASTRPC_LICENSE file included in this distribution.
+ *
+ *************************************************************************/
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperationv1.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperationv1.stg [hunk 15]
```diff
@@ -0,0 +1,768 @@
...
+
+#include "$ctx.filename$.h"
+#include <$ctx.includePrefix$protocols/dds/MessageHeader.h>
+#include <$ctx.includePrefix$utils/Messages.h>
+
+#include <cstddef>
+
+#if defined(_WIN32)
+#if defined(EPROSIMA_USER_DLL_EXPORT)
+#define user_cpp_DllExport __declspec( dllexport )
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperationv1.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterfacev1.stg (hunk 1 above)
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterfacev1.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperationv1.stg (hunk 2 above)
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperationv1.stg]: Done in 44e1b980

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TypesCdrAuxHeaderImpl.stg (hunk 3 above)
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TypesCdrAuxHeaderImpl.stg]: Done in 44e1b980

File: include/transports/dds/RTPSProxyTransport.h (hunk 5 above)
Comment (richiware) [reply include/transports/dds/RTPSProxyTransport.h]: Done in 44e1b980

File: include/transports/dds/components/RTPSServerProcedureEndpoint.h (hunk 6 above)
Comment (richiware) [reply include/transports/dds/components/RTPSServerProcedureEndpoint.h:27]: I agree with you. But we think is out of scope of this PR.

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperation.stg [hunk 16]
```diff
@@ -1,10 +1,16 @@
-/*************************************************************************
- * Copyright (c) 2013 eProsima. All rights reserved.
- *
- * This copy of FASTRPC is licensed to you under the terms described in the
- * FASTRPC_LICENSE file included in this distribution.
- *
- *************************************************************************/
+// Copyright 2013 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (MiguelCompany) [inline fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperation.stg:1]:
```suggestion
// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterface.stg [hunk 17]
```diff
@@ -1,10 +1,16 @@
-/*************************************************************************
- * Copyright (c) 2012 eProsima. All rights reserved.
- *
- * This copy of FASTRPC is licensed to you under the terms described in the
- * FASTRPC_LICENSE file included in this distribution.
- *
- *************************************************************************/
+// Copyright 2012 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (MiguelCompany) [inline fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterface.stg:1]:
```suggestion
// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperation.stg [hunk 18]
```diff
@@ -1,10 +1,16 @@
-/*************************************************************************
- * Copyright (c) 2012 eProsima. All rights reserved.
- *
- * This copy of FASTRPC is licensed to you under the terms described in the
- * FASTRPC_LICENSE file included in this distribution.
- *
- *************************************************************************/
+// Copyright 2012 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (MiguelCompany) [inline fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperation.stg:1]:
```suggestion
// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: fastrpcgen/src/com/eprosima/fastrpc/idl/parser/typecode/StructTypeCode.java [hunk 19]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+// Copyright 2046 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/parser/typecode/StructTypeCode.java]: Done in 36f24497

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByInterface.stg [hunk 20]
```diff
@@ -1,10 +1,16 @@
-/*************************************************************************
- * Copyright (c) 2013 eProsima. All rights reserved.
- *
- * This copy of FASTRPC is licensed to you under the terms described in the
- * FASTRPC_LICENSE file included in this distribution.
- *
- *************************************************************************/
+// Copyright 2013 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByInterface.stg]: Done in 36f24497

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperation.stg (hunk 16 above)
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsHeaderByOperation.stg]: Done in 36f24497

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterface.stg (hunk 17 above)
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByInterface.stg]: Done in 36f24497

File: fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperation.stg (hunk 18 above)
Comment (richiware) [reply fastrpcgen/src/com/eprosima/fastrpc/idl/templates/TopicsSourceByOperation.stg]: Done in 36f24497

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**